aten/src/ATen/core/jit_type.h

```diff
@@ -1708,4 +1710,55 @@ struct CAFFE2_API InterfaceType : public NamedType {
   bool is_module_;
 };
 
+template <TypeKind K>
+struct EnumerationType : public Type {
+  static const TypeKind Kind = K;
+
+  bool operator==(const Type& rhs) const override {
+    return rhs.kind() == kind();
+  }
+
+ protected:
+  EnumerationType() : Type(Kind) {}
+};
+
+struct LayoutType;
+using LayoutTypePtr = std::shared_ptr<LayoutType>;
+// This type represents a Generator
+struct CAFFE2_API LayoutType : public EnumerationType<TypeKind::LayoutType> {
+static LayoutTypePtr create() {
+  return LayoutTypePtr(
+      new LayoutType()); // NOLINT(modernize-make-shared)
```

Comment:
nit: use `make_shared`

Reply:
I will do this in a separate diff and fix it for all types listed here.